pyomo/contrib/doe/fim_doe_tutorial.ipynb

```diff
@@ -0,0 +1,2017 @@
+      " 300r-3.6443625e+01 1.51e+02 1.47e+03  -9.0 3.81e+04    -  1.12e-01 5.72e-04f  1\n",
+      " 301r-3.6496275e+01 4.76e+02 1.34e+03  -9.0 4.36e+04    -  5.77e-02 1.59e-01f  1\n",
+      " 302r-3.6601805e+01 1.25e+03 9.64e+02  -9.0 5.37e+04    -  1.43e-01 2.47e-01F  1\n",
+      " 303r-3.6820970e+01 4.54e+03 6.10e+02  -9.0 7.52e+04    -  7.51e-01 3.68e-01f  1\n",
+      " 304r-3.6898003e+01 4.44e+03 5.53e+02  -9.0 1.12e+05    -  4.41e-01 9.35e-02h  1\n",
+      " 305r-3.6898121e+01 4.43e+03 1.26e+03  -9.0 9.16e+04    -  2.48e-01 2.46e-03h  1\n",
+      " 306r-3.6907097e+01 4.26e+03 7.18e+02  -9.0 9.95e+04    -  6.74e-01 6.67e-02h  1\n",
+      " 307r-3.6907035e+01 4.25e+03 1.22e+03  -9.0 7.72e+04    -  1.79e-01 2.57e-03h  1\n",
+      " 308r-3.6907440e+01 4.27e+03 9.96e+02  -9.0 8.24e+04    -  9.18e-02 1.84e-01h  1\n",
+      " 309r-3.7112659e+01 1.55e+04 2.17e+02  -9.0 9.36e+04    -  2.19e-01 7.82e-01h  1\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      " 310r-3.7115232e+01 1.54e+04 3.52e+02  -9.0 7.47e+04    -  3.38e-01 7.66e-03h  1\n",
+      " 311r-3.7217331e+01 1.16e+04 3.25e+02  -9.0 7.61e+04    -  4.06e-01 2.90e-01h  1\n",
+      " 312r-3.7216737e+01 1.15e+04 2.17e+02  -9.0 9.97e+03    -  1.00e+00 3.07e-03h  1\n",
+      " 313r-3.7042091e+01 3.09e+02 4.41e+01  -9.0 1.27e+04    -  1.40e-01 1.00e+00h  1\n",
+      " 314r-3.7030355e+01 6.70e+00 1.19e-04  -9.0 5.60e+02    -  1.00e+00 1.00e+00h  1\n",
+      " 315r-3.7030270e+01 1.25e-03 1.13e-08  -9.0 6.23e+00    -  1.00e+00 1.00e+00h  1\n",
+      "\n",
+      "Number of Iterations....: 315\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:  -3.7030270449687649e+01   -3.7030270449687649e+01\n",
+      "Dual infeasibility......:   1.7384418534153198e-01    1.7384418534153198e-01\n",
+      "Constraint violation....:   1.2496497759308944e-03    1.2496497759308944e-03\n",
+      "Complementarity.........:   9.0909090910114887e-10    9.0909090910114887e-10\n",
+      "Overall NLP error.......:   1.2496497759308944e-03    1.7384418534153198e-01\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 707\n",
+      "Number of objective gradient evaluations             = 24\n",
+      "Number of equality constraint evaluations            = 711\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 320\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 316\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =     31.992\n",
+      "Total CPU secs in NLP function evaluations           =      1.349\n",
+      "\n",
+      "EXIT: Converged to a point of local infeasibility. Problem may be infeasible.\n",
```

Comment:
Ipopt didn't converge. Is this expected?

Reply:
Yes... This problem can get this non-converging message for the first solve, but it converges at the second solve which is really solving some degree of freedom, so we thought this is acceptable. What do you think?